### DISCUSSION TITLE
Fixedsys size locking

### DISCUSSION THREAD

**Discussion by gmardom** [budlabs/i3ass#121] [Q&A]
I was trying to make `Fixedsys` locked at `12pt`. I looked it up and what i came up with is
```xml
<match target="font">
  <test name="family">
    <string>Fixedsys Excelsior</string>
  </test>
  <edit name="size" mode="assign">
    <int>12</int>
  </edit>
</match>
```
Which doesn't work...

I tried to find a video where you showed it but I cannot find it after 2h of scrubbing videos.
Can you share how you do it? It would be so helpful.

**Comment by budRich** [comment]
yeah its possible i never made a video about this. The issue being that Fixedsys does not have a free lisence and I actually use a modified version of the font. Also in font config, you will need to add size overridez for both pixel and point values. (and possibly bold and italic options, since they look awful with fixedsys).

I am pretty sure i have written down the process needed to  modify the font, the modifications are needed to get the font to be truly monospaced, and it will look a lot better, especially in terminals when the modifications are applied. I will dig in my old blogarchive and see if i can find the article I wrote about it and post it if i find it. but below is my `fontconfig/conf.d/10-fixedsys.conf`  

```
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
<fontconfig>
  <match target="pattern">
    <test qual="any" name="family">
        <string>FixedFixedsys</string>
    </test>
    <test name="weight" compare="more">
        <const>medium</const>
    </test>
    <edit name="weight" mode="assign" binding="same">
        <const>medium</const>
    </edit>
  </match>

  <match target="pattern">
    <test qual="any" name="family"  compare="eq">
        <string>Fixedsys Excelsior 3.01-L2 Mono</string>
    </test>
    <test name="weight" compare="more">
        <const>medium</const>
    </test>
    <edit name="weight" mode="assign" binding="same">
        <const>medium</const>
    </edit>
  </match>


  <!-- disable anti-alias -->
  <match target="font">
    
    <test qual="any" name="family"  compare="eq">
        <string>FixedFixedsys</string>
    </test>
    
    <edit name="antialias" mode="assign">
      <bool>false</bool>
    </edit>

  </match>


  <match target="font">
    
    <test qual="any" name="family" compare="eq">
      <string>Fixedsys Excelsior 3.01-L2 Mono</string>
    </test>
    
    <edit name="antialias" mode="assign">
      <bool>false</bool>
    </edit>

  </match>

  <!-- fixed sizes -->
  <!-- polybar, cairo? -->
  <match target="font">

    <test qual="any" name="family" compare="eq">
      <string>FixedFixedsys</string>
    </test>

    <test qual="any" name="size" compare="not_eq">
      <double>11</double>
    </test>

    <edit name="size" mode="assign">
      <double>11</double>
    </edit>

  </match>

  <!-- works in GTK -->
  <match target="font">

    <test qual="any" name="family" compare="eq">
      <string>FixedFixedsys</string>
    </test>

    <test qual="any" name="pixelsize" compare="not_eq">
      <double>16</double>
    </test>

    <edit name="pixelsize" mode="assign">
      <double>16</double>
    </edit>

  </match>


</fontconfig>
```

**Reply by budRich** [reply]
FixedFixedsys is btw that modified version. And **Fixedsys Excelsior 3.01-L2 Mono** is the fixedsys version i use, but i am not sure it is available any more, i know there is a fixedsys with ligatures and stuff.. yeah i will research this a bit. I guess i could do a video similar to how i did with the amiga mouse cursor hack. I can show how to do it, but not redistribute the files so to speak.

**Reply by gmardom** [reply]
Video on that would be awesome dude.

Also, there is aur package for fixedsys font.
```
ttf-fixedsys-excelsior-linux
```

**Reply by gmardom** [reply]
Thanks for quick response. That worked!

**Comment by budRich** [comment]
I found my old blogpost. I will leave it here for historians, the things I describe in it may or may not work anymore, 5 has passed since I wrote it and my linux beard was only a moustache back then:

--------------------

I'm not sure why, but [Fixedsys](http://www.fixedsysexcelsior.com) is one of my all time favorite fonts. I particularly like it as a monospaced terminal and programming font (*Fixedsys is the font used on this site to display code and ASCII art*). But it hasn't been painless to get the terminal to accept Fixedsys.  

The problem from the start is that the available version of Fixedsys is a `.ttf` font. After some digging around the Internet, I found [this article on askubuntu.com](https://askubuntu.com/questions/210283/how-to-use-fixedsys-in-the-gnome-terminal-or-wherever-monospaced-fonts-are-requ).

I followed the instructions and managed to get the font in the terminal.

*but*

The font wasn't recognized in `.Xresources`. I had to put this dirthack in my `.bashrc`:

  
`printf '\e]710;%s\007' "xft:Fixedsys Excelsior 3.01\-L2 Mono:style=Regular"`  

 

Sure, it worked, but the hack became the cause of some strange, undesired behavior. 

But I couldn't give up, not when I was so close. So I did some experimenting, and I think the issue of `.Xresources` not recognizing the font, was the font name. So following the instructions in the [askubuntu.com article](https://askubuntu.com/questions/210283/how-to-use-fixedsys-in-the-gnome-terminal-or-wherever-monospaced-fonts-are-requ), but renaming the font to `FixedFixedsys` did the trick!

* * * * * * *

Below is a summary of the [askubuntu.com article](https://askubuntu.com/questions/210283/how-to-use-fixedsys-in-the-gnome-terminal-or-wherever-monospaced-fonts-are-requ) with some added screenshots and the name change. I wanted to save it here, and i guess the same technique can be used to enable other fonts.

* * * * * * *

Before anything else, make sure [fontforge](http://fontforge.org/) is installed. And get [this modified version of Fixedsys](https://bugs.launchpad.net/ubuntu/+bug/200671/comments/16)

Open `FSEX301-L2.ttf` with `fontforge`.


In the dialog asking to load bitmap fonts, select: `No`  

Open the menu `File → Execute Script...` (`Ctrl+.`) and paste the following code in the input box:
``` text
SelectAll()
foreach
   if (GlyphInfo("Width") != 80)
      DetachAndRemoveGlyphs()
   else
      b = GlyphInfo("BBox")
      if (b[0] < 0 || b[1] < -30 || b[2] > 80 || b[3] > 130)
         DetachAndRemoveGlyphs()
      endif
   endif
endloop
SelectNone()
```
Select `FF` and click `OK` (*it will take a while, you will have to wait until the Execute Script window disappears*)


Next open `Element → Font Info...` menu In PS Names change the Fontname, Family Name, and Name for Humans to `FixedFixedsys`.


In the same Font Information window, select OS/2.
In the Misc. tab change the PFM Family to Monospace.


In the Metrics tab, set Win Descent Offset and HHead Descent Offset both to 0 (*).


In the Panose tab, change Proportion to Monospaced.


In the same Font Information window, select Lookups.
In the GSUB tab, and in the GPOS tab, select each line in turn and click `Delete`.


In the same Font Information window, click `OK`, and in the Change UniqueID? window that appears click `Change`.


Open the `Element → Bitmap Strikes Available...` (`Ctrl+Shift+B`) menu. In the window that appears set Pixel Sizes to **16** and click `OK`.


Select `File → Generate Fonts...` menu (`Ctrl+Shift+G`), choose a different directory and/or a different filename if needed, make sure the output format is TrueType, and click `Generate`. In the Non-standard Em-Size window that appears click `Yes`. 


Also click `Generate` in the next window that appears.


Quit FontForge `File → Quit` (`Ctrl+Q`). In the Font changed window that appears click `Don't Save`.

Move the generated file to the `~/.fonts` directory (create it if it doesn't exist).

Now you should be able to set the font in `.Xresources` .

I would also suggest that you add the following rule to `~/.fonts.conf` to disable boldness from the font. (*it will render weirdly otherwise, and Fixedsys is, imo, bold enough as it is.*)

``` xml
<match target="pattern">
    <test qual="any" name="family">
        <string>FixedFixedsys</string>
    </test>
    <test name="weight" compare="more">
        <const>medium</const>
    </test>
    <edit name="weight" mode="assign" binding="same">
        <const>medium</const>
    </edit>
</match>
```

**Comment by budRich** [comment]
here are some links related to this:

https://askubuntu.com/a/210284 , the guied I followed to make it truely monospaced
http://web.archive.org/web/20160226120117/http://www.fixedsysexcelsior.com/ , archived version of original fixedsys excelsior page  
https://bugs.launchpad.net/ubuntu/+bug/200671 - discussion with links to several updated versions of fixedsys, the last links from 2019 appears to contain "mirrored" versions for bold variants, which if installed and used probably elliminates the need for those options in fontconfig.  
https://github.com/kika/fixedsys - the repo that the AUR now grabs the font from, somehow it is "public domain"...
https://en.wikipedia.org/wiki/Fixedsys - wikipedia, notice that the font is  "owned" by microsoft...
https://aur.archlinux.org/packages/ttf-fixedsys-excelsior-linux/ - AUR package

**Comment by budRich** [comment]
made a video of this
https://youtu.be/hmOrm3pkPNY
